geode-for-redis/src/main/java/org/apache/geode/redis/internal/commands/executor/set/SetOpExecutor.java

```diff
@@ -14,185 +14,54 @@
-import org.apache.geode.redis.internal.data.RedisSet;
-import org.apache.geode.redis.internal.data.RedisSet.MemberSet;
 import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
 import org.apache.geode.redis.internal.services.RegionProvider;
 
 public abstract class SetOpExecutor implements CommandExecutor {
 
   @Override
   public RedisResponse executeCommand(Command command, ExecutionHandlerContext context) {
-    int setsStartIndex = 1;
-
-    if (isStorage()) {
-      setsStartIndex++;
-    }
-
-    List<RedisKey> commandElements = command.getProcessedCommandKeys();
-    List<RedisKey> setKeys = commandElements.subList(setsStartIndex, commandElements.size());
     RegionProvider regionProvider = context.getRegionProvider();
+    List<RedisKey> commandElements = command.getProcessedCommandKeys();
+    List<RedisKey> setKeys = commandElements.subList(1, commandElements.size());
 
-    /*
-     * SINTERSTORE currently use the else part of the code for their implementation.
-     * TODO: Once the above commands have been implemented remove the if else
-     * Refactor so the implementation is in the executor. After delete doActualSetOperation,
-     * doStoreSetOp, doStoreSetOpWhileLocked, computeStoreSetOp, fetchSets
-     */
-    List<RedisKey> keysToLock = new ArrayList<>(setKeys);
-    if (isStorage()) {
-      keysToLock.add(command.getKey());
-    }
-    if (command.isOfType(RedisCommandType.SDIFF) || command.isOfType(RedisCommandType.SDIFFSTORE)) {
-      if (isStorage()) {
-        RedisKey destinationKey = command.getKey();
-        int resultSize = context.lockedExecute(destinationKey, keysToLock,
-            () -> sdiffstore(regionProvider, destinationKey, setKeys));
-        return RedisResponse.integer(resultSize);
-      }
-
-      Set<byte[]> resultSet = context.lockedExecute(setKeys.get(0), keysToLock,
-          () -> sdiff(regionProvider, setKeys));
-      return RedisResponse.array(resultSet, true);
-
-    } else if (command.isOfType(RedisCommandType.SINTER)) {
-      Set<byte[]> resultSet = context.lockedExecute(setKeys.get(0), keysToLock,
-          () -> sinter(regionProvider, setKeys));
-      return RedisResponse.array(resultSet, true);
-
-    } else if (command.isOfType(RedisCommandType.SUNION)
-        || command.isOfType(RedisCommandType.SUNIONSTORE)) {
-      if (isStorage()) {
-        RedisKey destinationKey = command.getKey();
-        int resultSize = context.lockedExecute(destinationKey, keysToLock,
-            () -> sunionstore(regionProvider, destinationKey, setKeys));
-        return RedisResponse.integer(resultSize);
-      }
-
-      Set<byte[]> resultSet = context.lockedExecute(setKeys.get(0), keysToLock,
-          () -> sunion(regionProvider, setKeys));
-      return RedisResponse.array(resultSet, true);
-    }
+    RedisResponse result = context.lockedExecute(setKeys.get(0), new ArrayList<>(setKeys),
+        () -> performCommand(regionProvider, setKeys));
```

Comment:
This can just be:
```
return context.lockedExecute(setKeys.get(0), new ArrayList<>(setKeys),
        () -> performCommand(regionProvider, setKeys));
```